src/iter/fold.rs

```diff
@@ -114,4 +118,8 @@ impl<'r, C, ID, F, T> Folder<T> for FoldFolder<'r, C, ID, F>
     fn complete(self) -> C::Result {
         self.base.consume(self.item).complete()
     }
+
+    fn full(&self) -> bool {
+        self.base.full()
+    }
```

Comment:
I view it as somewhere in between. For example, if you have an infinite iterator and you are using `find`, then if `full` doesn't work properly, your program might never terminate.